File: ios/SignLanguageTranslation/Network/SignModel.swift
// ios/SignLanguageTranslation/Network/SignModel.swift

import Foundation

struct SignModel: Codable {
    let state: Bool?
    let baseUrl: String?
    let name: String?
    let cid: String?
    let st: Bool?
    
    enum CodingKeys: String, CodingKey {
        case state
        case baseUrl = "baseUrl"
        case name
        case cid
        case st
    }
    
    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        
        state = try container.decodeIfPresent(Bool.self, forKey: .state)
        baseUrl = try container.decodeIfPresent(String.self, forKey: .baseUrl)
        name = try container.decodeIfPresent(String.self, forKey: .name)
        cid = try container.decodeIfPresent(String.self, forKey: .cid)
        st = try container.decodeIfPresent(Bool.self, forKey: .st)
    }
    
    init(state: Bool?, baseUrl: String?, name: String?, cid: String?, st: Bool?) {
        self.state = state
        self.baseUrl = baseUrl
        self.name = name
        self.cid = cid
        self.st = st
    }
    
    var videoUrl: String? {
        guard let baseUrl = baseUrl, let name = name else { return nil }
        let url = "\(baseUrl)\(name)"
        return url.replacingOccurrences(of: "http://", with: "https://")
    }
}

File: android/src/main/java/com/signlanguagetranslation/network/SignModel.kt
// android/src/main/java/com/signlanguagetranslation/network/SignModel.kt

package com.signlanguagetranslation.network

import org.json.JSONObject

data class SignModel(
    val state: Boolean?,
    val baseUrl: String?,
    val name: String?,
    val cid: String?,
    val st: Boolean?
) {
    companion object {
        fun fromJson(json: String): SignModel {
            val jsonObject = JSONObject(json)
            return SignModel(
                state = if (jsonObject.has("state")) jsonObject.getBoolean("state") else null,
                baseUrl = jsonObject.optString("baseUrl", null),
                name = jsonObject.optString("name", null),
                cid = jsonObject.optString("cid", null),
                st = if (jsonObject.has("st")) jsonObject.getBoolean("st") else null
            )
        }
    }

    val videoUrl: String?
        get() {
            if (baseUrl == null || name == null) return null
            val url = "$baseUrl$name"
            return url.replace("http://", "https://")
        }
}
